informer/informer.go

```diff
@@ -250,105 +248,6 @@ func (i *Informer) Watch(ctx context.Context) (chan watch.Event, chan watch.Even
 	return deleteChan, updateChan, errChan
 }
 
-// cacheAndSend stores the provided event object in the informer cache and
-// dispatches it based on its properties. cacheAndSend sends the provided event
-// object to the provided update channel in case the event object has no
-// deletion timestamp. In case the deletion timestamp of the provided event
-// object is non-nil, it is send to the provided delete channel.
-func (i *Informer) cacheAndSend(event watch.Event, deleteChan, updateChan chan watch.Event) error {
-	k, err := cache.MetaNamespaceKeyFunc(event.Object)
-	if err != nil {
-		return microerror.Mask(err)
-	}
-	i.cache.Store(k, event)
```

Comment:
Most of this is moved. What changed in the method is that I removed the caching from here because we do it in a more separate and dedicated way which makes the approach easier and cleaner and also fixes the issues we perceived.